### PR TITLE
scripts: tests: Blackbox test expansion - filter

### PR DESCRIPTION
**Pull request by KamilxPaszkiet**
Adds tests related to filter flags:
--arch
--vendor

### CODE REVIEW

File: scripts/tests/twister_blackbox/test_filter.py [hunk 1]
```diff
@@ -122,3 +182,69 @@ def test_enable_slow_only(self, out_path):
...
+            'vendor nxp',
+            'vendor intel',
+            'vendor ite'
+        ],
+    )
+
+    @mock.patch.object(TestPlan, 'TESTSUITE_FILENAME', testsuite_filename_mock)
+    def test_vendor(self, capfd, out_path, vendor, expected):
+        path = os.path.join(TEST_DATA, 'tests', 'no_filter')
+        test_platforms = ['qemu_x86','intel_ish_5_4_1','frdm_k64f','it8xxx2_evb']
```
Comment (LukaszMrugala) [inline scripts/tests/twister_blackbox/test_filter.py:232]: Add spaces after commas

**Review by LukaszMrugala — CHANGES_REQUESTED** [review]

File: scripts/tests/twister_blackbox/test_data/tests/no_filter/dummy/src/main.c [hunk 2]
```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2023 Intel Corporation
```
[the remark this answers is not in the document]
Comment (KamilxPaszkiet) [reply scripts/tests/twister_blackbox/test_data/tests/no_filter/dummy/src/main.c]: done

File: scripts/tests/twister_blackbox/test_filter.py [hunk 3]
```diff
@@ -122,3 +182,69 @@ def test_enable_slow_only(self, out_path):
...
+            'arch x86',
+            'arch arm',
+            'arch riscv'
+        ],
+    )
+
+    @mock.patch.object(TestPlan, 'TESTSUITE_FILENAME', testsuite_filename_mock)
+    def test_arch(self, capfd, out_path, arch, expected):
+        path = os.path.join(TEST_DATA, 'tests', 'no_filter')
+        test_platforms = ['qemu_x86','intel_ish_5_4_1','frdm_k64f','it8xxx2_evb']
```
[the remark this answers is not in the document]
Comment (KamilxPaszkiet) [reply scripts/tests/twister_blackbox/test_filter.py]: done

File: scripts/tests/twister_blackbox/test_filter.py (hunk 1 above)
Comment (KamilxPaszkiet) [reply scripts/tests/twister_blackbox/test_filter.py]: done

**Review by mdubielx — APPROVED** [review]

**Review by LukaszMrugala — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**